controllers/platform/services.go

```diff
@@ -0,0 +1,360 @@
+
+	databaseNamespace := platform.Namespace
+	if len(platform.Spec.Services.Persistence.PostgreSql.ServiceRef.Namespace) > 0 {
+		databaseNamespace = platform.Spec.Services.Persistence.PostgreSql.ServiceRef.Namespace
+	}
+	dataSourcePort := 5432
+	if platform.Spec.Services.Persistence.PostgreSql.ServiceRef.Port != nil {
+		dataSourcePort = *platform.Spec.Services.Persistence.PostgreSql.ServiceRef.Port
+	}
+	databaseName := "sonataflow"
+	if len(platform.Spec.Services.Persistence.PostgreSql.DatabaseName) > 0 {
+		databaseName = platform.Spec.Services.Persistence.PostgreSql.DatabaseName
+	}
+	dataSourceUrl := "jdbc:" + persistenceType + "://" + platform.Spec.Services.Persistence.PostgreSql.ServiceRef.Name + "." + databaseNamespace + ":" + strconv.Itoa(dataSourcePort) + "/" + databaseName + "?currentSchema=data-index-service"
+
+	secretRef := corev1.LocalObjectReference{
+		Name: platform.Spec.Services.Persistence.PostgreSql.SecretRef.Name,
+	}
+	quarkusDatasourceUsername := "POSTGRESQL_USER"
+	if len(platform.Spec.Services.Persistence.PostgreSql.SecretRef.UserKey) > 0 {
+		quarkusDatasourceUsername = platform.Spec.Services.Persistence.PostgreSql.SecretRef.UserKey
+	}
+	quarkusDatasourcePassword := "POSTGRESQL_PASSWORD"
+	if len(platform.Spec.Services.Persistence.PostgreSql.SecretRef.PasswordKey) > 0 {
+		quarkusDatasourcePassword = platform.Spec.Services.Persistence.PostgreSql.SecretRef.PasswordKey
+	}
+	persistEnvVars := []corev1.EnvVar{
+		{
+			Name: "QUARKUS_DATASOURCE_USERNAME",
+			ValueFrom: &corev1.EnvVarSource{
+				SecretKeyRef: &corev1.SecretKeySelector{
+					Key:                  quarkusDatasourceUsername,
+					LocalObjectReference: secretRef,
+				},
+			},
+		},
+		{
+			Name: "QUARKUS_DATASOURCE_PASSWORD",
+			ValueFrom: &corev1.EnvVarSource{
+				SecretKeyRef: &corev1.SecretKeySelector{
+					Key:                  quarkusDatasourcePassword,
+					LocalObjectReference: secretRef,
+				},
+			},
+		},
+		{
+			Name:  "QUARKUS_DATASOURCE_DB_KIND",
+			Value: persistenceType,
+		},
+		{
+			Name:  "QUARKUS_HIBERNATE_ORM_DATABASE_GENERATION",
+			Value: "update",
+		},
+		{
+			Name:  "QUARKUS_FLYWAY_MIGRATE_AT_START",
+			Value: "true",
+		},
+		{
+			Name:  "QUARKUS_DATASOURCE_JDBC_URL",
+			Value: dataSourceUrl,
+		},
+	}
+	dataDeployContainer.Env = append(dataDeployContainer.Env, persistEnvVars...)
+}
+
+func createDataIndexService(ctx context.Context, client client.Client, platform *operatorapi.SonataFlowPlatform) error {
+	lbl := map[string]string{
+		workflowproj.LabelApp: platform.Name,
+	}
+	dataSvcSpec := corev1.ServiceSpec{
+		Ports: []corev1.ServicePort{
+			{
+				Name:       httpPort,
+				Protocol:   corev1.ProtocolTCP,
+				Port:       80,
+				TargetPort: common.DefaultHTTPWorkflowPortIntStr,
+			},
+		},
+		Selector: lbl,
+	}
+	dataSvc := &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: platform.Namespace,
+			Name:      platform.Name + "-" + common.DataIndexName,
```

Comment:
`platform.Name + "-" + common.DataIndexName`  repeats several times.
perhaps replace it with a util function to determine the format of the name for all occurrences, since it is used both for creating the resource and referring to it.

Reply:
done